htdocs/comm/propal/class/propal.class.php

```diff
@@ -734,6 +735,10 @@ public function addline($desc, $pu_ht, $qty, $txtva, $txlocaltax1 = 0.0, $txloca
 				// Reorder if child line
 				if (!empty($fk_parent_line)) {
 					$this->line_order(true, 'DESC');
+				} elseif ($ranktouse > 0 && $ranktouse <= count($this->lines)) { // Update all rank of all other lines
+					for ($ii = $ranktouse; $ii <= count($this->lines); $ii++) {
```

Comment:
`$linecount = count($this->lines);
for ($ii = $ranktouse; $ii <= $linecount; $ii++) {`

Reply:
@frederic34 you were right, it's ok now with my PR, thank you very much ! :)